### ISSUE TITLE
View Owner's user mapping not used for VIEWs referencing foreign tables

### ISSUE THREAD

**Issue by proddata** [crate/crate#16040]
### CrateDB version

5.7.1

### CrateDB setup information

- cloud or local 
- single node

### Problem description

When trying to perform a SELECT operation on a view created by `user_a` using `user_b`, a PSQLException error occurs, indicating that password-based authentication was requested but no password was provided. The expectation is that the view should use the user and therefore the user mapping of the owner of the view, which is `user_a`.

This issue specifically pertains to views on foreign tables. For normal CrateDB tables, the permissions of the view owner are correctly used.

In PostgreSQL, when a user creates a view, the view is executed with the permissions of the user who owns the view. This means that any underlying foreign tables or queries should use the credentials and user mappings of the view's owner. 

```sql
georg=> SELECT current_user;
 current_user 
--------------
 georg2

georg=> SELECT * FROM x;
ERROR:  user mapping not found for "georg2"

georg=> SELECT * FROM x_v;
 classification | country | first_ascent | height |    mountain    | prominence |    range     |        region        
----------------+---------+--------------+--------+----------------+------------+--------------+----------------------
 I/B-09.V-A     | CH      |         1859 |   4199 | Rimpfischhorn  |        642 | Valais       | Mischabel
```

### Steps to Reproduce

1. **As `user_a`:**
    ```sql
    -- Create user mapping for user_a
    CREATE USER MAPPING FOR user_a ...;

    -- Create a foreign table in the schema doc
    CREATE FOREIGN TABLE doc.x (
       "region" TEXT
    ); 

    -- Create a view in the schema doc
    CREATE VIEW doc.x_v AS 
    SELECT * 
    FROM doc.x 
    WHERE region = 'y';

    -- Verify the view works for user_a
    SELECT * FROM doc.x_v;
    ```

    The above `SELECT` statement should work correctly and return the expected results.

2. **As `user_b`:**
    ```sql
    -- Attempt to query the view created by user_a
    SELECT * FROM doc.x_v;
    ```

### Actual Result

The following error is encountered:

```text
PSQLException[The server requested password-based authentication, but no password was provided by plugin null]
```

### Expected Result

The view `doc.x_v` should execute using the credentials and user mapping of `user_a`, the owner of the view.

**Comment by jeeminso** [comment]
Thanks for reporting @proddata. I should check why `MissingPrivilegeException` was not thrown.

**Comment by proddata** [comment]
@jeeminso the issue is not the returned exception, but rather that this should work and return results for `user_b` even though that user does not have access to the (foreign) table directly nor a user mapping.

> https://cratedb.com/docs/crate/reference/en/latest/general/ddl/views.html
> Privileges on relations accessed by the view are not necessary.
> However, it is required, at all times, that the owner (the user who created the view), has DQL privileges on all relations occurring within the view’s query definition.

**Comment by jeeminso** [comment]
Thanks again @proddata the fix will be available in `5.7.2` release.
